src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,29 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    WSingletonContainer::defineSingleton(child_group->objectName(),
+                                         child_group);
+}
+
+QWidget* LegacySkinParser::parseSingletonContainer(QDomElement node) {
+    if (!node.hasAttribute("objectName")) {
+        qWarning() << "Need objectName attribute for Singleton tag.";
+        return new QWidget();
+    }
+    QString objectName = node.attribute("objectName");
+    WSingletonContainer* pContainer =
+            WSingletonContainer::getSingleton(objectName, m_pParent);
+    if (pContainer == NULL) {
+        qWarning() << "ERROR: Singleton" << objectName
+                   << "not found, probably about to crash.";
+    }
+    commonWidgetSetup(node, pContainer);
```

Comment:
In the error case above I should just return new QWidget()

A single widget does not have two parents.  There are two WidgetGroups created, however.  One is the container, which is often empty (this is the actual WSingletonContainer).  The other is the WidgetGroup that is the singleton, created in the defineSingleton call and owned by the singletoncontainer.  This other group is the one that is reparented around, inserted in containers as necessary.